airflow/providers/amazon/aws/log/cloudwatch_task_handler.py

```diff
@@ -81,7 +81,7 @@ def set_context(self, ti):
         self.handler = watchtower.CloudWatchLogHandler(
             log_group=self.log_group,
             stream_name=self._render_filename(ti, ti.try_number),
-            boto3_session=self.hook.get_session(self.region_name),
+            boto3_client=self.hook.get_conn(),
```

Comment:
Watchtower now expects a Boto3 client instead of a session, this was changed in this commit: https://github.com/o-nikolas/watchtower/commit/a6232505d5aba2f5d2a67641e30bfbaa5874a0fa